### PR TITLE
#3 handle accents and a few cosmetic changes

### PR DESCRIPTION
**Pull request by fabmars**
#3 handle accents and a few cosmetic changes

### CODE REVIEW

File: src/test/java/fr/raluy/chocoratage/KeysBufferTest.java [hunk 1]
```diff
@@ -30,15 +33,52 @@ public void shouldAddStringToEmptyBuffer() {
...
+        addStringToBuffer("บพ"); // Thai
+        addStringToBuffer("ങജ"); // Malayalam
+        addStringToBuffer("ছয়"); // Bengali
+        addStringToBuffer("ఞఌ"); // Telugu
+        assertThat(keysBuffer.toString()).isEqualTo("laαψжцթխجؼשמ葵袷ばふ하함腌盒บพങജছয়ఞఌ");
+    }
+
+    @Test
+    public void shouldNotAddNonLetterOrDigit() {
+        addStringToBuffer("♥♥♥");
```
Comment (Draluy) [inline src/test/java/fr/raluy/chocoratage/KeysBufferTest.java:71]: Shouldnt we care about unicode chars depicting chocolatines?

**Review by Draluy — COMMENTED** [review]
You changed a lot of stuff. Give me some days to review everything plox <3

File: src/test/java/fr/raluy/chocoratage/KeysBufferTest.java (hunk 1 above)
Comment (fabmars) [reply src/test/java/fr/raluy/chocoratage/KeysBufferTest.java:71]: Yep, but imo now we can blacklist this char. I'll do a UT for this :)

**Comment by fabmars** [comment]
I didn't change much. The main change is the way keyCodes and keyChars are analyzed. What you were doing (getKeyText) was some sort of (unintentional?) workaround for the (poorly documented) way NativeKeyEvents are contextualized. Now we're as neat as Mr. Clean (plus I nearly have the same amount of hair).

**Comment by Draluy** [comment]
Hey, I cant build the project because I'm missing jnativehook-2.1.1-SNAPSHOT.jar
Could you provide it please?

**Comment by fabmars** [comment]
[jnativehook-2.1.1-SNAPSHOT.zip](https://github.com/Draluy/chocoratage/files/4187318/jnativehook-2.1.1-SNAPSHOT.zip)

**Comment by fabmars** [comment]
Unzip it,  github refuses to include jars (which are zips, go figure)
`mvn install:install-file -Dfile=/path/to/the/file/jnativehook-2.1.1-SNAPSHOT.jar -DgroupId=com.1stleg -DartifactId=jnativehook -Dversion=2.1.1-SNAPSHOT -Dpackaging=jar`

**Comment by Draluy** [comment]
Hi Fabien,

I'm on sick leave today and I plan on reviewing this PR. There are some things in the code that bothers me, amomgst them new README.

How open are you to changes? As you are the main contributor right now, I dont feel like its right for me to play the dictator, but on the other hand I can't host and maintain stuff I don't believe in.

**Comment by fabmars** [comment]
Huh sounds like serious matter. 
How do you mean? Did I write anything stupid or unnecessary?
For me it's a fun project, I'm learning new things, like some Unicode concepts or discovering the gradle far-west world. I don't plan on working on it much further than the improved matching ticket.

**Comment by Draluy** [comment]
Nothing serious, dont' worry :) I don't take myself or this project too seriously. Let me make some changes to your PR and submit them to you, it'll be easier.

File: src/main/java/fr/raluy/chocoratage/KeyBuffer.java [hunk 2]
```diff
@@ -37,49 +36,99 @@ public KeyBuffer(boolean strict) {
...
+    }
 
-        String keyText = NativeKeyEvent.getKeyText(keyEvent.getKeyCode());
+    /**
+     * http://www.physics.udel.edu/~watson/scen103/ascii.html
+     *
+     * @param keyChar
+     * @return true if the input char is in the range [0x00-0x1F] or DEL (0x7F) or space (0x20)
+     */
+    public static boolean isControlCharOrSpace(char keyChar) {
```
Comment (Draluy) [inline src/main/java/fr/raluy/chocoratage/KeyBuffer.java:126]: Is this safe to assume? Aren't we using UTF8 and not ascii?

File: src/main/java/fr/raluy/chocoratage/KeyLogger.java [hunk 3]
```diff
@@ -27,38 +27,44 @@ public KeyLogger() {
         }
     }
 
+    /**
+     * keyCode is defined here, not keyChar
```
Comment (Draluy) [inline src/main/java/fr/raluy/chocoratage/KeyLogger.java:31]: I dont understand what you meant here.

File: src/main/java/fr/raluy/chocoratage/Main.java [hunk 4]
```diff
@@ -2,30 +2,36 @@
...
+        }
 
         KeyLogger keyLogger = new KeyLogger();
         Os os = Optional.ofNullable(Config.getForcedOs()).orElseGet(() -> Os.guess());
+        SessionLocker sessionLocker = new SessionLocker(os);
+        keyLogger.addListener(sessionLocker);
+
         if (Config.isDebugMode()) {
             log.info("OS set to {}", os);
+            log.info("Forbidden phrases:", os);
```
Comment (Draluy) [inline src/main/java/fr/raluy/chocoratage/Main.java:32]: os gets logged twice?

**Comment by Draluy** [comment]
Here are the modifications I made:

- Rewrote README to be more to the point and neutral/serious in tone. 
- Separated french/english parts, because I doubt anyone will ever read the english part. I would personnally strip the english part altogheter. Is chocoblast even a thing abroad?
- Removed the liability section. Choosing a license such as GPLv3 will cover us.

- added the jnativehook lib in the project. Important to make the project work for anyone cloning the repo imho.

**Comment by Draluy** [comment]
I cannot get the program to work, be it using the command line or the IDE.
I get this 

`
Exception in thread "main" java.lang.UnsatisfiedLinkError: org.jnativehook.GlobalScreen.getAutoRepeatRate()Ljava/lang/Integer;
`
It does compile fine, I can see the jnativehook classes in the jar, but it still hangs on me.

When running the jar I get the error at this line:  GlobalScreen.registerNativeHook();

Any idea?

**Comment by Draluy** [comment]
Could you test it on a Debian VM? It seems to me this version of jnativehook crashes.

File: src/main/java/fr/raluy/chocoratage/KeyLogger.java (hunk 3 above)
Comment (fabmars) [reply src/main/java/fr/raluy/chocoratage/KeyLogger.java:31]: depending on the context EITHER keyCode (nativeKeyPressed, nativeKeyReleased) OR keyChar (nativeKeyTyped) is defined in the nativeKeyEvent object. It's not even the same object and it's confusing, so I commented on that to remember next time I come back in this area.

File: src/main/java/fr/raluy/chocoratage/Main.java (hunk 4 above)
Comment (fabmars) [reply src/main/java/fr/raluy/chocoratage/Main.java:32]: It's a bad case of copy paste, there's not even a `{}`. Should be `log.info("Forbidden phrases:");`

File: src/test/java/fr/raluy/chocoratage/KeysBufferTest.java (hunk 1 above)
Comment (fabmars) [reply src/test/java/fr/raluy/chocoratage/KeysBufferTest.java:71]: see #14

File: src/main/java/fr/raluy/chocoratage/KeyBuffer.java (hunk 2 above)
Comment (fabmars) [reply src/main/java/fr/raluy/chocoratage/KeyBuffer.java:126]: Yeah they are aligned up to 0x7F :)
Read the 2nd § of https://en.wikipedia.org/wiki/UTF-8

**Comment by fabmars** [comment]
Ah damn, I realize the jar I gave you was a quick compile on windows (yes I develop this on windows, sometimes Ubuntu shame on me). On any other OS you'll miss the native lib (.so...).
So, I need to compile jnativehook correctly on all platforms. It's quite a struggle, believe me.

**Comment by fabmars** [comment]
[jnativehook-2.1.20200213.zip](https://github.com/Draluy/chocoratage/files/4198248/jnativehook-2.1.20200213.zip)
There you are @Draluy. Very sorry again.

**Comment by Draluy** [comment]
Thx a lot :) And no need to apologize :)

**Pull request closed without merging**